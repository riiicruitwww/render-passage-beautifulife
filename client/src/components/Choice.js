import React from 'react';
import './Choice.scss';
import Paragraph from './Paragraph';

const Choice = ({ choice, questionId, userAnswer, onClickChoice }) => {
  const number = `(${String.fromCharCode(97 + choice.number)})`;
  const isSelected = userAnswer[questionId] === choice.number;

  const handleChoiceClick = (choiceNumber, ev) => {
    onClickChoice(questionId, choiceNumber);
  };

  const renderChoice = () => {
    return choice.children.map((item, index) => (
      <Paragraph key={index} paragraph={item} />
    ));
  };

  return (
    <div
      className={isSelected ? 'Choice selected' : 'Choice'}
      onClick={handleChoiceClick.bind(this, choice.number)}
    >
      <span className="Choice__number">{number}</span>
      {renderChoice()}
    </div>
  );
};

export default Choice;
