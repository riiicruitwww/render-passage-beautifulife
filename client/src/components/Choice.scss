.Choice {
  display: flex;
  cursor: pointer;

  &__number {
    margin-right: 1em;
  }

  &.selected {
    color: rgb(240, 40, 40);
  }

  p {
    margin-bottom: 0.3em;
  }
}
