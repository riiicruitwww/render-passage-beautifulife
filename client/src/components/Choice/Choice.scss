.Choice {
  display: flex;
  transition: all 0.3s;
  cursor: pointer;

  &__number {
    margin-right: 16px;
  }

  &.answer {
    color: #0000ff;
  }

  &.wrong {
    color: #f02828;
  }

  &.selected {
    color: #f02828;
  }

  p {
    margin-bottom: 0.3rem;
  }
}
