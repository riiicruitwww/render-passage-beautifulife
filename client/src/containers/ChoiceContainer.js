import { connect } from 'react-redux';
import { Choice } from 'Contents';
import { selectAnswerComplete, selectAnswerEvent } from 'actions';
import charCodeToNum from 'utils/charCodeToNum';

const mapStateToProps = (state, { choice, questionId }) => {
  const number = charCodeToNum(choice.number);
  const isSelected = state.ui.userAnswer[questionId] === number;
  let choiceType = isSelected ? 'selected' : '';

  if (state.ui.isCheckComplete) {
    if (state.ui.checkResult[questionId].correctAnswer === number) {
      choiceType = 'answer';
    } else if (isSelected) {
      choiceType = 'wrong';
    }
  }

  return {
    choiceType,
    number,
    questionsLength: state.content.questions.length,
    userAnswer: state.ui.userAnswer
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onComplete: () => {
      dispatch(selectAnswerComplete());
    },
    onSelect: (questionId, choiceNumber) => {
      dispatch(selectAnswerEvent(questionId, choiceNumber));
    }
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Choice);
