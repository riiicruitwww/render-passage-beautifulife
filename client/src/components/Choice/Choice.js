import React from 'react';
import './Choice.scss';
import Paragraph from '../Paragraph/Paragraph';

const Choice = ({
  choice,
  choiceType,
  number,
  questionId,
  questionsLength,
  userAnswer,
  onComplete,
  onSelect
}) => {
  const handleChoiceClick = () => {
    onSelect(questionId, number);

    if (
      questionsLength === Object.keys(userAnswer).length + 1 &&
      !userAnswer[questionId]
    ) {
      onComplete();
    }
  };

  const renderChoice = () => {
    return choice.children.map((item, index) => (
      <Paragraph key={index} paragraph={item} />
    ));
  };

  return (
    <div
      className={`Choice ${choiceType}`}
      onClick={handleChoiceClick}
    >
      <span className="Choice__number">{`(${number})`}</span>
      {renderChoice()}
    </div>
  );
};

export default Choice;
