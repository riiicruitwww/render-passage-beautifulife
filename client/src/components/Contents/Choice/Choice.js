import React from 'react';
import PropTypes from 'prop-types';
import './Choice.scss';
import { Paragraph } from 'Contents';

const Choice = ({
  choice,
  choiceType,
  number,
  questionId,
  questionsLength,
  userAnswer,
  onComplete,
  onSelect
}) => {
  const handleChoiceClick = () => {
    onSelect(questionId, number);

    if (
      questionsLength === Object.keys(userAnswer).length + 1 &&
      !userAnswer[questionId]
    ) {
      onComplete();
    }
  };

  const renderChoice = () => {
    return choice.children.map((item, index) => (
      <Paragraph key={index} paragraph={item} />
    ));
  };

  return (
    <div className={`Choice ${choiceType}`} onClick={handleChoiceClick}>
      <span className="Choice__number">{`(${number})`}</span>
      {renderChoice()}
    </div>
  );
};

Choice.propTypes = {
  choice: PropTypes.shape({
    children: PropTypes.array.isRequired
  }).isRequired,
  choiceType: PropTypes.string.isRequired,
  number: PropTypes.string.isRequired,
  questionId: PropTypes.number.isRequired,
  questionsLength: PropTypes.number.isRequired,
  userAnswer: PropTypes.objectOf(PropTypes.string).isRequired,
  onComplete: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired
};

export default Choice;
