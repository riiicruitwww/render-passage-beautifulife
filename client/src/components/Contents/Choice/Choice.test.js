import React from 'react';
import { shallow } from 'enzyme';
import ChoiceContainer from 'containers/ChoiceContainer';
import Choice from './Choice';

describe('Choice', () => {
  describe('about render', () => {
    const props = {
      choice: {
        children: []
      }
    };
    const wrapper = shallow(<Choice {...props} />);

    it('compare with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('component rendered', () => {
      expect(wrapper.find('.Choice').length).toBe(1);
    });
  });
});
