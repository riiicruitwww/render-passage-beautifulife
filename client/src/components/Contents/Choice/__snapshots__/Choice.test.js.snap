// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choice about render compare with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Choice
    choice={
      Object {
        "children": Array [],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="Choice__number"
        >
          (undefined)
        </span>,
        Array [],
      ],
      "className": "Choice undefined",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "(undefined)",
          "className": "Choice__number",
        },
        "ref": null,
        "rendered": "(undefined)",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="Choice__number"
          >
            (undefined)
          </span>,
          Array [],
        ],
        "className": "Choice undefined",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "(undefined)",
            "className": "Choice__number",
          },
          "ref": null,
          "rendered": "(undefined)",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
